### PR TITLE
feat: added timeout as attribute to checks

### PR DESCRIPTION
**Pull request by bartvdbraak**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

While working on Pull Request #830, I noticed that some checks were missing the `timeout` attribute. 

If you look at the [API request specification](https://learn.microsoft.com/en-us/rest/api/azure/devops/approvalsandchecks/check-configurations/add?view=azure-devops-rest-7.0&tabs=HTTP) and the GUI on Azure DevOps, it becomes clear that this is part of their spec and has a certain default and a max size of a 32-bit integer:

**Example cURL for Branch Control check**
```
curl \
  'https://dev.azure.com/{organization}/{project}/_apis/pipelines/checks/configurations?api-version=7.0-preview.1' \
  -X POST \
  --data-raw '
    {
      "type": {
        "id": "fe1de3ee-a436-41b4-bb20-f6eb4cb879a7",
        "name": "Task Check"
      },
      "settings": {
        "definitionRef": {
          "id": "86b05a0c-73e6-4f7d-b3cf-e38f3b39a75b",
          "name": "evaluatebranchProtection",
          "version": "0.0.1"
        },
        "displayName": "Branch control",
        "inputs": {
          "allowedBranches": "*",
          "ensureProtectionOfBranch": "false"
        },
        "retryInterval": 0,
        "linkedVariableGroup": null
      },
      "resource": {
        "type": "environment",
        "id": "{environmentId}",
        "name": "{environmentName}"
      },
      "timeout": 50000
    }'
```

**Example cURL for Business Hours check**
```
curl \
  'https://dev.azure.com/{organization}/{project}/_apis/pipelines/checks/configurations?api-version=7.0-preview.1' \
  -X POST \
  --data-raw '
    {
      "type": {
        "id": "fe1de3ee-a436-41b4-bb20-f6eb4cb879a7",
        "name": "Task Check"
      },
      "settings": {
        "definitionRef": {
          "id": "445fde2f-6c39-441c-807f-8a59ff2e075f",
          "name": "evaluateBusinessHours",
          "version": "0.0.1"
        },
        "displayName": "Business Hours",
        "inputs": {
          "businessDays": "Monday,Tuesday,Wednesday,Thursday,Friday",
          "timeZone": "UTC",
          "startTime": "04:00",
          "endTime": "11:00"
        },
        "retryInterval": 0,
        "linkedVariableGroup": null
      },
      "resource": {
        "type": "environment",
        "id": "{environmentId}",
        "name": "{environmentName}"
      },
      "timeout": 2147483647
    }'
```

**GUI Examples in Azure DevOps**
Screenshot 2023-07-20 at 00 18 04
Screenshot 2023-07-20 at 00 18 44

Issue Number:

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [x] No

## Does this introduce a breaking change?

- [x] No

## Any relevant logs, error output, etc?

n/a

## Other information

n/a

### CODE REVIEW

File: azuredevops/internal/acceptancetests/resource_check_branch_control_test.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func TestAccCheckBranchControl_basic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(tfCheckNode, "project_id"),
 					resource.TestCheckResourceAttr(tfCheckNode, "allowed_branches", branches),
 					resource.TestCheckResourceAttr(tfCheckNode, "display_name", checkName),
+					resource.TestCheckResourceAttr(tfCheckNode, "timeout", "50000"),
```
Comment (xuzhang3) [inline azuredevops/internal/acceptancetests/resource_check_branch_control_test.go]: default value is `1440` not `5000`

File: azuredevops/internal/acceptancetests/resource_check_branch_control_test.go [hunk 2]
```diff
@@ -56,6 +57,7 @@ func TestAccCheckBranchControl_complete(t *testing.T) {
...
```
Comment (xuzhang3) [inline azuredevops/internal/acceptancetests/resource_check_branch_control_test.go]: default value is `1440` not `5000`

File: azuredevops/internal/acceptancetests/resource_check_business_hours_test.go [hunk 3]
```diff
@@ -78,6 +78,7 @@ func TestAccCheckBusinessHours_complete(t *testing.T) {
 					resource.TestCheckResourceAttr(tfCheckNode, "friday", "false"),
 					resource.TestCheckResourceAttr(tfCheckNode, "saturday", "false"),
 					resource.TestCheckResourceAttr(tfCheckNode, "sunday", "false"),
+					resource.TestCheckResourceAttr(tfCheckNode, "timeout", "50000"),
```
Comment (xuzhang3) [inline azuredevops/internal/acceptancetests/resource_check_business_hours_test.go]: default is `1440` not `5000`

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
Please ensure all the AccTests passed

File: azuredevops/internal/acceptancetests/resource_check_business_hours_test.go (hunk 3 above)
Comment (bartvdbraak) [reply azuredevops/internal/acceptancetests/resource_check_business_hours_test.go]: What about the values that are used in the Acceptance tests for "Check Approval"?
See:
https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L33

https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L59

File: azuredevops/internal/acceptancetests/resource_check_branch_control_test.go (hunk 2 above)
Comment (bartvdbraak) [reply azuredevops/internal/acceptancetests/resource_check_branch_control_test.go]: What about the values that are used in the Acceptance tests for "Check Approval"?
See:
https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L33

https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L59

File: azuredevops/internal/acceptancetests/resource_check_branch_control_test.go (hunk 1 above)
Comment (bartvdbraak) [reply azuredevops/internal/acceptancetests/resource_check_branch_control_test.go]: What about the values that are used in the Acceptance tests for "Check Approval"?
See:
https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L33

https://github.com/microsoft/terraform-provider-azuredevops/blob/bd2d69d460e62ea22f0d11252d58268bd5c16130/azuredevops/internal/acceptancetests/resource_check_approval_test.go#L59

**Comment by bartvdbraak** [comment]
> Please ensure all the AccTests passed

Can you show me what is failing in the Acceptance tests? 

Currently I see that all checks have passed:
Screenshot 2023-07-26 at 14 16 00

**Comment by xuzhang3** [comment]
@bartvdbraak  The  AccTest has been fixed by the latest commit. The pipeline only run the unit tests not AccTests.
Error log
```log
=== CONT  TestAccCheckBranchControl_complete
    resource_check_branch_control_test.go:48: Step 1/1 error: Check failed: Check 5/7 error: azuredevops_check_branch_control.test: Attribute 'timeout' expected "50000", got "1440"
=== CONT  TestAccCheckBranchControl_basic
    resource_check_branch_control_test.go:22: Step 1/1 error: Check failed: Check 5/5 error: azuredevops_check_branch_control.test: Attribute 'timeout' expected "50000", got "1440"

```

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccCheckBranchControl_basic
=== PAUSE TestAccCheckBranchControl_basic
=== RUN   TestAccCheckBranchControl_complete
=== PAUSE TestAccCheckBranchControl_complete
=== RUN   TestAccCheckBranchControl_update
=== PAUSE TestAccCheckBranchControl_update
=== CONT  TestAccCheckBranchControl_basic
=== CONT  TestAccCheckBranchControl_update
=== CONT  TestAccCheckBranchControl_complete
--- PASS: TestAccCheckBranchControl_complete (80.09s)
--- PASS: TestAccCheckBranchControl_basic (80.43s)
--- PASS: TestAccCheckBranchControl_update (101.69s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        108.609s

=== RUN   TestAccCheckBusinessHours_basic
=== PAUSE TestAccCheckBusinessHours_basic
=== RUN   TestAccCheckBusinessHours_complete
=== PAUSE TestAccCheckBusinessHours_complete
=== RUN   TestAccCheckBusinessHours_update
=== PAUSE TestAccCheckBusinessHours_update
=== CONT  TestAccCheckBusinessHours_basic
=== CONT  TestAccCheckBusinessHours_complete
=== CONT  TestAccCheckBusinessHours_update
--- PASS: TestAccCheckBusinessHours_complete (98.87s)
--- PASS: TestAccCheckBusinessHours_basic (98.93s)
--- PASS: TestAccCheckBusinessHours_update (128.72s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        130.944s
```

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**